source/common/http/utility.h

```diff
@@ -101,7 +101,7 @@ namespace Utility {
  */
 class Url {
 public:
-  bool initialize(absl::string_view absolute_url);
+  bool initialize(absl::string_view absolute_url, bool is_connect_request = false);
```

Comment:
nit: I would probably have this be a non-default param given how much it changes the behavior, and possibly even an enum for clarity, but up to you.